src/app/profile/edit/form/index.tsx

```diff
@@ -0,0 +1,102 @@
+"use client";
+import { Input } from "@/components/input";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import z from "zod";
+import { editSchema } from "./constants";
+import { Button } from "@/components/button";
+import { useCurrentUser, useUpdateUser } from "@/domains/user/hooks";
+import { pickBy } from "lodash";
+import { ErrorHelpers } from "@/services/error/helpers";
+import { toast } from "react-toastify";
+
+type EditFormData = z.infer<typeof editSchema>;
+
+const EditUserForm = () => {
+  const { data } = useCurrentUser();
+  const user = data?.data;
+
+  const mutation = useUpdateUser();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+    reset,
+  } = useForm<EditFormData>({
+    resolver: zodResolver(editSchema),
+    defaultValues: user,
+  });
+
+  const onSubmit = handleSubmit(async (data) => {
+    try {
+      const filteredData = pickBy(data, (value) => value.length > 0);
```

Comment:
There might be some fields that may be empty, so you should save them with empty state. It's better to not pick these values